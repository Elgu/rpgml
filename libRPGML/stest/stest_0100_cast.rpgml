#!/usr/bin/env rpgml

print( getType(   bool( 12.34 ) ) + "\n" ); 
print( getType(  int8 ( 12.34 ) ) + "\n" ); 
print( getType( uint8 ( 12.34 ) ) + "\n" ); 
print( getType(  int16( 12.34 ) ) + "\n" ); 
print( getType( uint16( 12.34 ) ) + "\n" ); 
print( getType(  int32( 12.34 ) ) + "\n" ); 
print( getType( uint32( 12.34 ) ) + "\n" ); 
print( getType(  int64( 12.34 ) ) + "\n" ); 
print( getType( uint64( 12.34 ) ) + "\n" ); 
print( getType( string( 12.34 ) ) + "\n" ); 

print( getArrayType(   bool( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType(  int8 ( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType( uint8 ( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType(  int16( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType( uint16( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType(  int32( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType( uint32( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType(  int64( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType( uint64( [ 12.34, 23.2 ] ) ) + "\n" ); 
print( getArrayType( string( [ 12.34, 23.2 ] ) ) + "\n" ); 

print( getArrayType(   bool( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType(  int8 ( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType( uint8 ( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType(  int16( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType( uint16( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType(  int32( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType( uint32( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType(  int64( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType( uint64( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 
print( getArrayType( string( [ [ 12.34 ], [ 23.2 ] ] ) ) + "\n" ); 

print( getArrayType(   bool( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType(  int8 ( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType( uint8 ( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType(  int16( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType( uint16( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType(  int32( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType( uint32( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType(  int64( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType( uint64( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 
print( getArrayType( string( [ [ 12.34 ], [ 23.2 ] ] )[ 0 ] ) + "\n" ); 

