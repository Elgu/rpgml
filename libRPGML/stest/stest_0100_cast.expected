bool
int8
uint8
int16
uint16
int32
uint32
int64
uint64
string
bool
int8
uint8
int16
uint16
int32
uint32
int64
uint64
string
other
other
other
other
other
other
other
other
other
other
bool
int8
uint8
int16
uint16
int32
uint32
int64
uint64
string
